### PR TITLE
perlPackages.AppMusicChordPro: init at 0.977

### PR DESCRIPTION
**Pull request by tdroxler**
###### Motivation for this change

[ChordPro](https://www.chordpro.org/chordpro/index.html) was missing
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/perl-packages.nix [hunk 1]
```diff
@@ -2487,6 +2500,21 @@ let
...
+  ChordPro = buildPerlPackage {
+    pname = "App-Music-ChordPro";
+    version = "0.974";
+    src = fetchurl {
+      url = mirror://cpan/authors/id/J/JV/JV/App-Music-ChordPro-0.974.tar.gz;
+      sha256 = "7fae50535dcb7f68171a8c1f8ad49b13563f406fb1d44eba78b349e7ef83a8af";
+    };
+    propagatedBuildInputs = [ AppPackager FileLoadLines FontTTF IOString ImageInfo PDFAPI2 StringInterpolateNamed ];
+    meta = {
+      homepage = http://www.chordpro.org;
```
Comment (stigtsp) [inline pkgs/top-level/perl-packages.nix:2512]:
```suggestion
      homepage = "http://www.chordpro.org";
```

Quote URLs per https://github.com/NixOS/rfcs/pull/45

File: pkgs/top-level/perl-packages.nix [hunk 2]
```diff
@@ -2487,6 +2500,21 @@ let
...
   };
 
+  ChordPro = buildPerlPackage {
+    pname = "App-Music-ChordPro";
+    version = "0.974";
+    src = fetchurl {
+      url = mirror://cpan/authors/id/J/JV/JV/App-Music-ChordPro-0.974.tar.gz;
+      sha256 = "7fae50535dcb7f68171a8c1f8ad49b13563f406fb1d44eba78b349e7ef83a8af";
+    };
+    propagatedBuildInputs = [ AppPackager FileLoadLines FontTTF IOString ImageInfo PDFAPI2 StringInterpolateNamed ];
```
Comment (stigtsp) [inline pkgs/top-level/perl-packages.nix:2510]:
```suggestion
    buildInputs = [ PodParser ];
    propagatedBuildInputs = [ AppPackager FileLoadLines IOString ImageInfo PDFAPI2 StringInterpolateNamed TextLayout Wx ];
```
`PodParser` is needed for compatibility with `perl-5.32.0`
`Wx` can be added for `bin/wxchordpro` support
`FontTTF` is already included by `PDFAPI2`
`TextLayout` is also needed (by 0.977), here is the derivation:

```nix
  TextLayout = buildPerlPackage {
    pname = "Text-Layout";
    version = "0.019";
    src = fetchurl {
      url = "mirror://cpan/authors/id/J/JV/JV/Text-Layout-0.019.tar.gz";
      sha256 = "a043f2a89e113b29c523a9efa71fa8398ed75edd482193901b38d08dd4a4108e";
    };
    buildInputs = [ PDFAPI2 ];
    meta = {
      description = "Pango style markup formatting";
      license = with stdenv.lib.licenses; [ artistic1 gpl1Plus ];
    };
  };
```

**Review by stigtsp — CHANGES_REQUESTED** [review]
Hi @tdroxler and welcome to the NixOS project!

`0.977` is available on CPAN, so please update to the latest version. I've added a few comments to this review.

**Comment by tdroxler** [comment]
Hi @stigtsp thx for the review and the help.

I fixed the comments and I already squashed/rebased so it's clean.

Cheers

**Comment by stigtsp** [comment]
Great, thx! Can you also change the commit/PR title from `perlPackages.ChordPro: init at 0.977` to `perlPackages.AppMusicChordPro: init at 0.977` so it matches the attribute?

**Comment by stigtsp** [comment]
Also, can you move `AppMusicChordPro` so it's in alphabetical order in `perl-packages.nix`?

**Comment by tdroxler** [comment]
done

**Comment by stigtsp** [comment]
@GrahamcOfBorg build perlPackages.AppMusicChordPro

File: pkgs/top-level/perl-packages.nix [hunk 3]
```diff
@@ -597,6 +597,35 @@ let
     };
   };
 
+  AppMusicChordPro = buildPerlPackage {
+    pname = "App-Music-ChordPro";
+    version = "0.977";
+    src = fetchurl {
+      url = mirror://cpan/authors/id/J/JV/JV/App-Music-ChordPro-0.977.tar.gz;
```
Comment (stigtsp) [inline pkgs/top-level/perl-packages.nix:604]:
```suggestion
      url = "mirror://cpan/authors/id/J/JV/JV/App-Music-ChordPro-0.977.tar.gz";
```

File: pkgs/top-level/perl-packages.nix [hunk 4]
```diff
@@ -597,6 +597,35 @@ let
...
+      sha256 = "0ggip43cddi5f6rylb07f56dhkfhbcbm621lvcnjfadnn9lrbwqh";
+    };
+    buildInputs = [ PodParser ];
+    propagatedBuildInputs = [ AppPackager FileLoadLines IOString ImageInfo PDFAPI2 StringInterpolateNamed TextLayout Wx ];
```
Comment (stigtsp) [inline pkgs/top-level/perl-packages.nix:608]: The following can be added to support darwin

```suggestion
    propagatedBuildInputs = [ AppPackager FileLoadLines IOString ImageInfo PDFAPI2 StringInterpolateNamed TextLayout ]
      ++ stdenv.lib.optional (!stdenv.isDarwin) [ Wx ];
    nativeBuildInputs = stdenv.lib.optional stdenv.isDarwin shortenPerlShebang;
    postInstall = stdenv.lib.optionalString stdenv.isDarwin ''
      shortenPerlShebang $out/bin/chordpro
      rm $out/bin/wxchordpro # Wx not supported on darwin
    '';
```

**Review by stigtsp — CHANGES_REQUESTED** [review]
Found two more things:
- url for `AppMusicChordPro` should be quoted
- Breaks on Darwin due to: Wx not supporting the platform, and `bin/chordpro` needs `shortenPerlShebang`

File: pkgs/top-level/perl-packages.nix (hunk 3 above)
Comment (tdroxler) [reply pkgs/top-level/perl-packages.nix]: done

File: pkgs/top-level/perl-packages.nix (hunk 4 above)
Comment (tdroxler) [reply pkgs/top-level/perl-packages.nix]: done, thx again for the help.

File: pkgs/top-level/perl-packages.nix [hunk 5]
```diff
@@ -597,6 +597,28 @@ let
...
+      sha256 = "0ggip43cddi5f6rylb07f56dhkfhbcbm621lvcnjfadnn9lrbwqh";
+    };
+    buildInputs = [ PodParser ];
+    propagatedBuildInputs = [ AppPackager FileLoadLines IOString ImageInfo PDFAPI2 StringInterpolateNamed TextLayout ]
+        ++ stdenv.lib.optional (!stdenv.isDarwin) [ Wx ];
+      nativeBuildInputs = stdenv.lib.optional stdenv.isDarwin shortenPerlShebang;
+      postInstall = stdenv.lib.optionalString stdenv.isDarwin ''
+        shortenPerlShebang $out/bin/chordpro
+        rm $out/bin/wxchordpro # Wx not supported on darwin
+      '';
```
Comment (stigtsp) [inline pkgs/top-level/perl-packages.nix:614]: Sorry, the indentation on these lines are a bit off. Should align with `propagatedBuildInputs` above.

Comment (tdroxler) [reply pkgs/top-level/perl-packages.nix]: np, it's done

**Comment by stigtsp** [comment]
@GrahamcOfBorg build perlPackages.AppMusicChordPro perldevelPackages.AppMusicChordPro

**Review by stigtsp — APPROVED** [review]
Looks good to me, thx for the contribution

Build and tested OK on `linux x64` and `darwin x64`

Result of `nixpkgs-review pr 100295` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>10 packages built:</summary>
  <ul>
    <li>perl530Packages.AppMusicChordPro</li>
    <li>perl530Packages.AppPackager</li>
    <li>perl530Packages.FileLoadLines</li>
    <li>perl530Packages.StringInterpolateNamed</li>
    <li>perl530Packages.TextLayout</li>
    <li>perl532Packages.AppMusicChordPro</li>
    <li>perl532Packages.AppPackager</li>
    <li>perl532Packages.FileLoadLines</li>
    <li>perl532Packages.StringInterpolateNamed</li>
    <li>perl532Packages.TextLayout</li>
  </ul>
</details>

**Comment by tdroxler** [comment]
thx for your help.

**Pull request merged**